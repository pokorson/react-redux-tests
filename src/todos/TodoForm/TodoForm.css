.todo-form {
  border: 1px solid #bbb;
  padding: 10px;
}
