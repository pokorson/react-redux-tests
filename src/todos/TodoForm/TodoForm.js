import React from 'react';
import './TodoForm.css';

class TodoForm extends React.PureComponent {
  state = {
    title: '',
  };

  updateTitle = e => {
    this.setState({
      title: e.target.value,
    });
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.onSubmit({
      title: this.state.title,
    });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit} className='todo-form'>
        <div>
          <label>title:</label>
          <input type='text' name='title' value={this.state.title} onChange={this.updateTitle} />
        </div>
        <button onClick={this.handleSubmit}>submit</button>
      </form>
    );
  }
}

export default TodoForm;
