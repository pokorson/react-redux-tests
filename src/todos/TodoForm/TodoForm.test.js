import React from 'react';
import TodoForm from './TodoForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TodoForm component', () => {
  it('renders component', () => {
    const component = shallow(<TodoForm />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('calls onSubmit props on button click', () => {
    const onSubmit = jest.fn();
    const component = shallow(<TodoForm onSubmit={onSubmit} />);
    component.find('button').simulate('click', { preventDefault: () => {} });
    expect(onSubmit).toHaveBeenCalled();
  });

  it('updates title on input change', () => {
    const component = shallow(<TodoForm />);
    const newTitle = 'new test title';
    component.find('input[name="title"]').simulate('change', { target: { value: newTitle } });
    expect(component.instance().state).toHaveProperty('title', newTitle);
  });

  it('renders title inside input', () => {
    const component = shallow(<TodoForm />);
    const title = 'title for state';
    component.setState({ title });
    expect(component.find('input').props()).toHaveProperty('value', title);
    expect(toJson(component)).toMatchSnapshot();
  });
});
