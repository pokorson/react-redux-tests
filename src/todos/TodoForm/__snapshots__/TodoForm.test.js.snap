// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm component renders component 1`] = `
<form
  className="todo-form"
  onSubmit={[Function]}
>
  <div>
    <label>
      title:
    </label>
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
  >
    submit
  </button>
</form>
`;

exports[`TodoForm component renders title inside input 1`] = `
<form
  className="todo-form"
  onSubmit={[Function]}
>
  <div>
    <label>
      title:
    </label>
    <input
      name="title"
      onChange={[Function]}
      type="text"
      value="title for state"
    />
  </div>
  <button
    onClick={[Function]}
  >
    submit
  </button>
</form>
`;
